### PR TITLE
Mon 20411 boolean rules too slow

### PR DESCRIPTION
**Pull request by bouda1**
## Description

Several fixes, optimizations on boolean rules.

REFS: MON-20411

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

### CODE REVIEW

File: tests/bam/boolean_rules.robot [hunk 1]
```diff
@@ -193,6 +193,170 @@ BABOOAND
...
+    Config Engine    ${1}
+
+    Clone Engine Config To DB
+    Add Bam Config To Engine
+    Add Bam Config To Broker    central
+    Set Services Passive    ${0}    service_302
+    Set Services Passive    ${0}    service_303
+    Set Services Passive    ${0}    service_304
+
+    ${id_ba__sid}=    create_ba    boolean-ba    impact    70    80
```
Comment (thearchit3ct) [inline tests/bam/boolean_rules.robot:219]: Can be improved by using Title Case in the test suite.
```suggestion
    ${id_ba__sid}    Create Ba    boolean-ba    impact    70    80
```

File: tests/bam/boolean_rules.robot [hunk 2]
```diff
@@ -193,6 +193,170 @@ BABOOAND
...
+    ${id_bool}=    add_boolean_kpi
```
Comment (thearchit3ct) [inline tests/bam/boolean_rules.robot:220]: There's no need for equality for all return value in the test suite.
```suggestion
    ${id_bool}    Add Boolean Kpi
```

File: tests/bam/boolean_rules.robot [hunk 3]
```diff
@@ -193,6 +193,170 @@ BABOOAND
...
+    ...    False
+    ...    100
+
+    Start Broker
+    ${start}=    Get Current Date
+    Start Engine
+    # Let's wait for the external command check start
+    ${content}=    Create List    check_for_external_commands()
+    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    msg=A message telling check_for_external_commands() should be available.
```
Comment (thearchit3ct) [inline tests/bam/boolean_rules.robot:232]:
```suggestion
    Should Be True    ${result}    A message telling check_for_external_commands() should be available.
```

File: tests/bam/boolean_rules.robot [hunk 4]
```diff
@@ -193,6 +193,170 @@ BABOOAND
...
+    ...    {host_16 service_302} {IS} {OK} {OR} {host_16 service_303} {IS} {OK}
+    ...    False
+    ...    100
+
+    Start Broker
+    ${start}=    Get Current Date
+    Start Engine
+    # Let's wait for the external command check start
+    ${content}=    Create List    check_for_external_commands()
+    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
```
Comment (thearchit3ct) [inline tests/bam/boolean_rules.robot:231]:
```suggestion
    ${result}=    Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
```

File: tests/bam/boolean_rules.robot [hunk 5]
```diff
@@ -193,6 +193,170 @@ BABOOAND
...
+
+    # Let's wait for the external command check start
+    ${content}=    Create List    check_for_external_commands()
+    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    msg=A message telling check_for_external_commands() should be available.
+    ${result}=    check_ba_status_with_timeout    boolean-ba    2    30
+    Should Be True    ${result}    msg=The 'boolean-ba' BA is not CRITICAL as expected
+
+    Stop Engine
+    Kindly Stop Broker
```
Comment (thearchit3ct) [inline tests/bam/boolean_rules.robot:292]: To be sure, the Keywords are executed even if the test fails.
```suggestion
    [Teardown]  Run Keywords  Stop Engine  AND  Kindly Stop Broker
```

**Review by thearchit3ct — APPROVED** [review]

**Review by jean-christophe81 — APPROVED** [review]

**Pull request merged**